cmake/PythonModuleCheck.cmake

```diff
@@ -0,0 +1,28 @@
+#=========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# Function will check that the specified modules are available to the python interpreter
+# If they are not cmake will throw an error indicating which module not available
+function(required_python_modules_found module_list)
+  if(Python_Interpreter_FOUND)
+    foreach(module IN LISTS module_list )
+      execute_process(COMMAND ${Python3_EXECUTABLE} -c "import ${module}"
+        RESULT_VARIABLE IMPORT_MODULE ERROR_QUIET)
+    
+      if(${IMPORT_MODULE} EQUAL 1)
+        message(FATAL_ERROR "Required python module ${module} was not found.") 
```

Comment:
Could you instead aggregate a list of all the missing modules and print them out as a single error message? That way you find out what you're missing all at once.

Reply:
Sure.